src/snowflake/cli/plugins/nativeapp/teardown_processor.py

```diff
@@ -89,9 +97,28 @@ def drop_application(self, auto_yes: bool):
                     cc.message(f"Did not drop application object {self.app_name}.")
                     return  # The user desires to keep the app, therefore exit gracefully
 
-        # 4. All validations have passed, drop object
+        # 4. If cascade was not explicitly set, check for objects owned by the app
+        if cascade is None:
```

Comment:
How do `cascade` and `auto_yes` (populated by `--force`) interact? What would `--force --no-cascade` do?

Reply:
To give background `--force` was added so that we could cleanup the app / app pkg even if they weren't created by Snowflake CLI or they had a different owner than expected.

Reply:
The behavior will be:
`--force`: cascade
`--force --no-cascade`: no cascade

Reply:
This was my suggestion, but now I'm remembering that you can teardown without cascading to the objects (I thought it blocked it). Yeah we should not have `--force` imply `--cascade`

Reply:
My bad @sfc-gh-gbloom sorry about that

Reply:
👍